openupgrade_scripts/scripts/analytic/17.0.1.2/pre-migration.py

```diff
@@ -1,12 +1,29 @@
 # Copyright 2024 Viindoo Technology Joint Stock Company (Viindoo)
 # Copyright 2024 Hunki enterprises - Holger Brunn
-# Copyright 2024 Tecnativa - Pedro M. Baeza
+# Copyright 2024,2025 Tecnativa - Pedro M. Baeza
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 from openupgradelib import openupgrade
 
 
 def _fill_config_parameter_analytic_project_plan(env):
-    env["ir.config_parameter"].set_param("analytic.project_plan", "1")
+    """If the system parameter is already set (because we have pre-filled according our
+    needs), we don't touch it.
+
+    If not, we check if there's already an existing ir.model.data entry for projects
+    according standard data (also pre-filled externally).
+
+    Finally, if not found, we get the first existing plan.
+    """
+    if env["ir.config_parameter"].get_param("analytic.project_plan", False):
+        return
+    imd = env["ir.model.data"].search(
+        [("module", "=", "analytic"), ("name", "=", "analytic_plan_projects")]
+    )
+    plan_id = imd.res_id
+    if not plan_id:
+        env.cr.execute("SELECT min(id) FROM account_analytic_plan")
```

Comment:
@pedrobaeza 
there is a regression on this line.
If there is no `account_analytic_plan`  it returns nothing 


I then got an issue 
```
  File "/odoo/src/addons/analytic/models/analytic_plan.py", line 107, in __get_all_plans
    project_plan = self.browse(int(self.env['ir.config_parameter'].sudo().get_param('analytic.project_plan', 0)))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: invalid literal for int() with base 10: 'None'

```

I don't get why I got None instead of 0 in get_param

Maybe 
```sql
SELECT COALESCE(min(id),0) min  FROM account_analytic_plan;
```
Is a bit better because we expect plan_id to have a value in line 26

Reply:
Solved other way in #4921. At the end, we will always end up with an existing `analytic.analytic_plan_projects` record sooner or later, so we have to stick to it.